src/main/java/Parser.java

```diff
@@ -0,0 +1,121 @@
+import java.util.LinkedList;
+
+public class Parser {
+    
+    public static LinkedList<String> parseAndExecute(String input, TaskList tasks, FileHandler fileHandler) 
+            throws DukeException {
+        //Split input incase it has a command and argument
+        String[] inputArray = input.split(" ", 2);
+
+        String[] cmdArgs = new String[0];
+        Task newTask;
+
+        LinkedList<String> result = new LinkedList<>();
+
+        switch (inputArray[0]) { 
+        case "list":
+            Task currTask;
+
+            //Append to result
+            for (int i = 1; i <= tasks.getTaskListLength(); i++) {
+                currTask = tasks.getTask(i - 1);
+                result.add(i + ". " + currTask);
+            }
+
+            break;
+        case "done":
+            if (inputArray.length < 2) {
+                throw new DukeException("OOPS!!! You need to specify the number of the task "
+                        + "you want to mark as done!");
+            }
+            int taskNum = Integer.parseInt(inputArray[1]);
+            int taskIndex = taskNum - 1;
+
+            Task modifiedTask = tasks.getTask(taskIndex);
+            modifiedTask.markAsDone();
+
+            //Saving
+            fileHandler.saveTasks(tasks);

```

Comment:
Would it be better that the saving is outside of the switch case? So there will be lesser repeated lines.

Reply:
I like the idea, in this case, I was considering the possibility of other commands that might not modify the TaskList so it made more sense to me to explicitly call saveTasks() only for commands that needed it.